gandalf.js

```diff
@@ -1,117 +1,114 @@
 /**
```

Comment:
SCRIPT-I'm a bit concerned about the potential performance impact of this loop. Have you thought about optimizing it?